src/frontend-proxy.ts

```diff
@@ -0,0 +1,95 @@
+import { getSubdomain, getPathname } from './url-utils'
+
+export const FRONTEND_DOMAIN = 'frontend-sooty-ten.now.sh'
+const API_ENDPOINT = 'https://api.serlo.org/graphql'
+const FRONTEND_ALLOWED_TYPES = ['Article', 'TaxonomyTerm']
+const FRONTEND_PROBABILITY = 0.1
+
+export async function handleRequest(
+  request: Request,
+  probability = FRONTEND_PROBABILITY,
+  allowdTypes = FRONTEND_ALLOWED_TYPES
+): Promise<Response | null> {
+  const url = request.url
+  const path = getPathname(url)
+
+  if (getSubdomain(url) !== 'de') return null
+
+  if (path === '/enable-frontend')
+    return createResponse('Enable frontend', probability, true)
+  if (path === '/disable-frontend')
+    return createResponse('Disable frontend', probability, false)
+
+  if (path.startsWith('/_next/') || path.startsWith('/_asset/'))
+    return await fetchBackend(true, false)
+
+  const typename = await queryTypename(path)
+  if (typename === null || !allowdTypes.includes(typename)) return null
+
+  const cookies = request.headers.get('Cookie')
+
+  if (cookies?.includes(formatCookie(true, probability)))
+    return fetchBackend(true, false)
```

Comment:
probably more readable if you combine that into an object `{useFrontend: true, setCookie: false}`